jscomp/others/js_json.ml

```diff
@@ -52,6 +52,31 @@ let reifyType (type a) (x : 'a) : (a kind * a ) =
 
 let reify_type = reifyType 
 
+type tagged_t = 
+  | JSONFalse
+  | JSONTrue
+  | JSONNull
+  | JSONString of string 
+  | JSONNumber of float 
+  | JSONObject of t Js_dict.t   
+  | JSONArray of t array 
+
+let classify  (x : 'a) : tagged_t =
+  let ty = Js.typeof x in  
+  if ty = "string" then 
+    JSONString (Obj.magic x)
+  else if ty = "number" then 
+    JSONNumber (Obj.magic x )
+  else if ty = "boolean" then
+    if (Obj.magic x) == Js.true_ then JSONTrue
+    else JSONFalse else
```

Comment:
This last `else` is very oddly placed I think. It's also not consistent with the formatting earlier in the function.